File: 682-BaseballGame/682-BaseballGame.java
// Last updated: 5/30/2025, 7:34:29 PM
import java.util.ArrayList;
class Solution {
    public int calPoints(String[] operations) {
        int S = 0;
        List<Integer> record = new ArrayList<Integer>();
        for (int i=0;i<operations.length;i++) {
            int lastIndex = record.size() - 1;
            if (operations[i].equals("C")) {
                Integer poppedElement = record.remove(lastIndex);
            }
            else if (operations[i].equals("D")){
                record.add(record.get(lastIndex)*2);
            }
            else if (operations[i].equals("+")) {
                record.add(record.get(lastIndex)+record.get(lastIndex-1));
            }
            else {
                record.add(Integer.parseInt(operations[i]));
            }
        }
        
        for (int score : record) {
            S += score;
        }
        return S;
    }
}

File: 682-BaseballGame/682-BaseballGame.py
# Last updated: 6/11/2025, 7:20:13 PM
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        stack = []
        for i in operations:
            if i == 'C':
                stack.pop()
            elif i == 'D':
                stack.append(stack[-1] * 2)
            elif i == '+':
                stack.append(stack[-1] + stack[-2])
            else:
                stack.append(int(i))
        return sum(stack)